fs/remote/blob.go

```diff
@@ -64,15 +67,33 @@ type blob struct {
 	checkInterval     time.Duration
 	fetchTimeout      time.Duration
 
-	fetchedRegionSet   regionSet
-	fetchedRegionSetMu sync.Mutex
+	fetchedRegionSet    regionSet
+	fetchedRegionSetMu  sync.Mutex
+	fetchedRegionGroup  singleflight.Group
+	fetchedRegionCopyMu sync.Mutex
 
 	resolver *Resolver
 
 	closed   bool
 	closedMu sync.Mutex
 }
 
+func makeBlob(fetcher *fetcher, size int64, chunkSize int64, prefetchChunkSize int64,
+	blobCache cache.BlobCache, lastCheck time.Time, checkInterval time.Duration,
+	r *Resolver, fetchTimeout time.Duration) *blob {
+	return &blob{
+		fetcher:           fetcher,
+		size:              size,
+		chunkSize:         chunkSize,
+		prefetchChunkSize: prefetchChunkSize,
+		cache:             blobCache,
+		lastCheck:         lastCheck,
+		checkInterval:     checkInterval,
+		resolver:          r,
+		fetchTimeout:      fetchTimeout,
+	}
+}
+
```

Comment:
That function was born because of complexity to create `blob` object during previous operations. Even though the object creation complexity is back to normal, I would prefer to keep the function, since this allows us to extend `blob` without worrying about implementing `makeBlob` again.